src/main/java/net/dv8tion/jda/internal/requests/Requester.java

```diff
@@ -264,17 +275,13 @@ private void applyBody(Request<?> apiRequest, okhttp3.Request.Builder builder)
         builder.method(method, body);
     }
 
-    private void applyHeaders(Request<?> apiRequest, okhttp3.Request.Builder builder, boolean authorized)
+    private void applyHeaders(Request<?> apiRequest, okhttp3.Request.Builder builder)
     {
-        builder.header("user-agent", USER_AGENT)
+        builder.header("user-agent", userAgent)
                .header("accept-encoding", "gzip")
+               .header("authorization", authConfig.getToken())
```

Comment:
This actually makes me nervous. I don't like the idea of accidentally forgetting this for the requester and sending the token to non-discord.
Could we instead check to see if the request is using the `RestConfig.baseUrl` to attach this?

Reply:
No, since proxies use it too. Making requests to untrusted servers isn't really something we can prevent.